libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -680,41 +650,20 @@ private void updateBannerInstructions(BannerText primaryBanner, BannerText secon
     }
   }
 
-  private void showGuidanceView(float currentStepDistanceTraveled, double currentStepTotalDistance) {
-    int metersBeforeShowGuidanceView = 200;
-    if (currentStepTotalDistance <= metersBeforeShowGuidanceView) {
```

Comment:
Are all of the step distance checks not needed anymore? What has changed?

Reply:
Yeah banner instructions tell us when to display the guidance view. We were doing unnecessary calculations earlier. These are definitely not needed. The implementation with this PR is much more promising.
cc @LukasPaczos